### ISSUE TITLE
500 error "min() arg is an empty sequence" when requesting a thumbnail

### ISSUE THREAD

**Issue by darnir** [matrix-org/synapse#5117]
### Description

Looking at the logfile for my homeserver, I see the the following error message pop up fairly frequently:

```
2019-05-01 14:12:25,452 - synapse.http.server - 112 - ERROR - GET-38086 - Failed handle request via 'ThumbnailResource': <XForwardedForRequest at 0x7fe3dd9286a0 method='GET' uri='/_matrix/media/v1/thumbnail/raim.ist/KthgQjZpYLHEVOIzwvTjkXPx?width=150&height=150&method=scale' clientproto='HTTP/1.0' site=8008>
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
  File "/usr/lib/python3.6/site-packages/synapse/storage/_base.py", line 460, in runWithConnection
    defer.returnValue(result)
  File "/usr/lib/python3.6/site-packages/twisted/internet/defer.py", line 1362, in returnValue
    raise _DefGen_Return(val)
twisted.internet.defer._DefGen_Return: [{'thumbnail_width': 10, 'thumbnail_height': 10, 'thumbnail_method': 'crop', 'thumbnail_type': 'image/jpeg', 'thumbnail_length': 715, 'filesystem_id': 'ZsQNhbXXFdIhkmfWZvdvlAww'}]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
  File "/usr/lib/python3.6/site-packages/synapse/storage/_base.py", line 418, in runInteraction
    defer.returnValue(result)
  File "/usr/lib/python3.6/site-packages/twisted/internet/defer.py", line 1362, in returnValue
    raise _DefGen_Return(val)
twisted.internet.defer._DefGen_Return: [{'thumbnail_width': 10, 'thumbnail_height': 10, 'thumbnail_method': 'crop', 'thumbnail_type': 'image/jpeg', 'thumbnail_length': 715, 'filesystem_id': 'ZsQNhbXXFdIhkmfWZvdvlAww'}]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/synapse/http/server.py", line 81, in wrapped_request_handler
    yield h(self, request)
  File "/usr/lib/python3.6/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/usr/lib/python3.6/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/usr/lib/python3.6/site-packages/synapse/rest/media/v1/thumbnail_resource.py", line 83, in _async_render_GET
    width, height, method, m_type
  File "/usr/lib/python3.6/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
  File "/usr/lib/python3.6/site-packages/synapse/rest/media/v1/thumbnail_resource.py", line 244, in _respond_remote_thumbnail
    width, height, method, m_type, thumbnail_infos
  File "/usr/lib/python3.6/site-packages/synapse/rest/media/v1/thumbnail_resource.py", line 317, in _select_thumbnail
    return min(info_list2)[-1]
ValueError: min() arg is an empty sequence
```

There are 10s of such messages every day. 

### Steps to reproduce

I don't currently know what action can cause this error.

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- **Homeserver**: My own HS

If not matrix.org:

<!--
What version of Synapse is running?
You can find the Synapse version by inspecting the server headers (replace matrix.org with
your own homeserver domain):
$ curl -v https://matrix.org/_matrix/client/versions 2>&1 | grep "Server:"
-->
- **Version**:  0.99.3

- **Install method**:  pip3
<!-- examples: package manager/git clone/pip  -->

- **Platform**: 
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->

Running inside a Alpine Linux LXC container on a Arch Linux host

**Comment by darnir** [comment]
Any suggestions / hints on why I'm seeing this and what can be done about it?

**Comment by darnir** [comment]
Anything at all? At the very least an acknowledgement of the issue would be nice.

If there's more info required, please let me know

**Comment by richvdh** [comment]
> an acknowledgement of the issue would be nice.

sorry, not sure what I can say. Looks like an error. don't know what causes it, other than (obviously) it's related to thumbnails.

**Comment by richvdh** [comment]
possibly a dup of #3183

**Comment by darnir** [comment]
I don't think so. I still see such errors in my logs. I'm happy to help in any way I can to debug this.

**Comment by richvdh** [comment]
> possibly a dup of #3183

this was the wrong link. I think I meant #3763. Or possibly #3813? However, I don't think it is really a dup of either of those. The root cause seems to be that a scaled thumbnail has been requested, but no scaled thumbnail has been generated for that image.

**Comment by FlyveHest** [comment]
I have just installed a Synapse server from the latest docker image (matrixdotorg/synapse:latest), and I am experiencing the same thing.

> Feb  1 20:55:50 server docker/synapse[1746]: 2021-02-01 20:55:50,974 - synapse.http.server - 79 - ERROR - GET-13559 - Failed handle request via 'ThumbnailResource': <XForwardedForRequest at 0x7f309cfbeac0 method='GET' uri='/_matrix/media/r0/thumbnail/server.name/RdKqjaYyUAbouqMHCeVCbese?width=16&height=16&method=crop' clientproto='HTTP/1.1' site='8008'>
Feb  1 20:55:50 server docker/synapse[1746]: Traceback (most recent call last):
Feb  1 20:55:50 server docker/synapse[1746]:   File "/usr/local/lib/python3.8/site-packages/synapse/http/server.py", line 224, in _async_render_wrapper
Feb  1 20:55:50 server docker/synapse[1746]:     callback_return = await self._async_render(request)
Feb  1 20:55:50 server docker/synapse[1746]:   File "/usr/local/lib/python3.8/site-packages/synapse/http/server.py", line 252, in _async_render
Feb  1 20:55:50 server docker/synapse[1746]:     callback_return = await raw_callback_return
Feb  1 20:55:50 server docker/synapse[1746]:   File "/usr/local/lib/python3.8/site-packages/synapse/rest/media/v1/thumbnail_resource.py", line 74, in _async_render_GET
Feb  1 20:55:50 server docker/synapse[1746]:     await self._respond_local_thumbnail(
Feb  1 20:55:50 server docker/synapse[1746]:   File "/usr/local/lib/python3.8/site-packages/synapse/rest/media/v1/thumbnail_resource.py", line 111, in _respond_local_thumbnail
Feb  1 20:55:50 server docker/synapse[1746]:     thumbnail_info = self._select_thumbnail(
Feb  1 20:55:50 server docker/synapse[1746]:   File "/usr/local/lib/python3.8/site-packages/synapse/rest/media/v1/thumbnail_resource.py", line 352, in _select_thumbnail
Feb  1 20:55:50 server docker/synapse[1746]:     return min(crop_info_list2)[-1]
Feb  1 20:55:50 server docker/synapse[1746]: ValueError: min() arg is an empty sequence

**Comment by clokep** [comment]
This should be fixed by #9163 in Synapse v1.27.0. Forgot to link the issues.
